src/mscorlib/src/System/Threading/Timer.cs

```diff
@@ -40,23 +40,27 @@ namespace System.Threading
     // The data structure we've chosen is an unordered doubly-linked list of active timers.  This gives O(1) insertion
     // and removal, and O(N) traversal when finding expired timers.
     //
-    // Note that all instance methods of this class require that the caller hold a lock on TimerQueue.Instance.
+    // Note that all instance methods of this class require that the caller hold a lock on the TimerQueue instance.
     //
     internal class TimerQueue
     {
-        #region singleton pattern implementation
+        #region Shared TimerQueue instances
 
-        // The one-and-only TimerQueue for the AppDomain.
-        private static TimerQueue s_queue = new TimerQueue();
+        public static TimerQueue[] Instances { get; } = CreateTimerQueues();
 
-        public static TimerQueue Instance
+        private TimerQueue(int id)
         {
-            get { return s_queue; }
+            m_id = id;
         }
 
-        private TimerQueue()
+        private static TimerQueue[] CreateTimerQueues()
         {
-            // empty private constructor to ensure we remain a singleton.
+            var queues = new TimerQueue[Environment.ProcessorCount];
```

Comment:
>Environment.ProcessorCount [](start = 40, length = 26)

Is this guaranteed to exist in all OS's?

Reply:
ProcessorCount? Yes.

Reply:
Just wondering if for some reason someone deleted this environment variable what will happen? do we depend on that in other places? don't we have any other way to get this info?

Reply:
This property has nothing to do with environment variables.

Reply:
good to know that :-)